packages/synapse-react-client/mocks/faker/generateFakeEntity.ts

```diff
@@ -0,0 +1,84 @@
+import { faker } from '@faker-js/faker'
+import {
+  Entity,
+  FileEntity,
+  Folder,
+  Project,
+} from '@sage-bionetworks/synapse-types'
+import { mockFileHandle } from '../mock_file_handle'
+import { pickRandomMockUser } from './fakerUtils'
+
+// When automatically generating entities, only create IDs in this range
+const MOCK_GENERATED_ENTITY_ID_SPACE_MIN = 10000
+const MOCK_GENERATED_ENTITY_ID_SPACE_MAX = 99999
+
+// IDs in this range are always projects
+export const MOCK_PROJECT_ID_SPACE_MIN = 10000
+export const MOCK_PROJECT_ID_SPACE_MAX = 19999
+
+// Folders
+const MOCK_FOLDER_ID_SPACE_MIN = 20000
+const MOCK_FOLDER_ID_SPACE_MAX = 39999
+
+// Files
+const MOCK_FILE_ID_SPACE_MIN = 40000
+const MOCK_FILE_ID_SPACE_MAX = 79999
+
+// TODO: denote id space for each entity type
+
+function generateBaseEntity(id: number): Entity {
+  return {
+    id: `syn${id}`,
+    createdBy: String(pickRandomMockUser().id),
+    createdOn: faker.date.anytime().toISOString(),
+    etag: faker.string.uuid(),
+    modifiedBy: String(pickRandomMockUser().id),
+    modifiedOn: faker.date.anytime().toISOString(),
+    name: faker.lorem.words({ min: 1, max: 4 }),
+    concreteType: 'org.sagebionetworks.repo.model.FileEntity',
+  }
+}
+
+function generateFile(id: number, versionNumber = 1): FileEntity {
+  return {
+    ...generateBaseEntity(id),
+    versionNumber: versionNumber,
+    dataFileHandleId: mockFileHandle.id,
+    concreteType: 'org.sagebionetworks.repo.model.FileEntity',
+  }
+}
+
+function generateFolder(id: number): Folder {
+  return {
+    ...generateBaseEntity(id),
+    concreteType: 'org.sagebionetworks.repo.model.Folder',
+  }
+}
+
+function generateProject(id: number): Project {
+  return {
+    ...generateBaseEntity(id),
+    name: faker.lorem.words({ min: 1, max: 4 }),
+    concreteType: 'org.sagebionetworks.repo.model.Project',
+  }
+}
+export function generateEntity(id: number, versionNumber?: number): Entity {
+  // TODO: Generate an entire bundle and return the relevant part
+  // TODO: Come up with a strategy for a stable entity hierarchy structure
```

Comment:
It would be nice if we could generate a stable hierarchy, then generate loads of mock data for cases like the entity finder. That needs more future thought.